apiserver/facades/client/spaces/spaces.go

```diff
@@ -291,6 +317,73 @@ func (api *API) ListSpaces() (results params.ListSpacesResults, err error) {
 	return results, nil
 }
 
+func (api *APIv5) ShowSpace(_, _ struct{}) {}
+
+// ListSpaces lists all the available spaces and their associated subnets.
+func (api *API) ShowSpace(entities params.Entities) (params.ShowSpaceResults, error) {
+	canRead, err := api.authorizer.HasPermission(permission.ReadAccess, api.backing.ModelTag())
+	if err != nil && !errors.IsNotFound(err) {
+		return params.ShowSpaceResults{}, errors.Trace(err)
+	}
+	if !canRead {
+		return params.ShowSpaceResults{}, common.ServerError(common.ErrPerm)
+	}
+
+	err = api.checkSupportsSpaces()
+	if err != nil {
+		return params.ShowSpaceResults{}, common.ServerError(errors.Trace(err))
+	}
+	results := make([]params.ShowSpaceResult, len(entities.Entities))
+	for i, entity := range entities.Entities {
+		spaceName, err := names.ParseSpaceTag(entity.Tag)
+		if err != nil {
+			results[i].Error = common.ServerError(errors.Trace(err))
+			continue
+		}
+		var result params.ShowSpaceResult
+		space, err := api.backing.SpaceByName(spaceName.Id())
+		if err != nil {
+			results[i].Error = common.ServerError(errors.Trace(err))
+			continue
+		}
+		result.Space = params.Space{}
+		result.Space.Name = space.Name()
+		result.Space.Id = space.Id()
+		subnets, err := space.Subnets()
+		if err != nil {
+			err = errors.Annotatef(err, "fetching subnets")
```

Comment:
Nit: not a fan of this shadowing, you could just move this to the following line or make a new variable.